### PR TITLE
Schema.js database agnostic

### PR DESCRIPTION
**Pull request by flovilmart**
- Schema.js now handles the Parse Schema format in and out
- Removes _DONOTUSE functions

### CODE REVIEW

File: src/Controllers/DatabaseController.js [hunk 1]
```diff
@@ -90,9 +90,8 @@ DatabaseController.prototype.loadSchema = function(acceptor = returnsTrue) {
 DatabaseController.prototype.redirectClassNameForKey = function(className, key) {
   return this.loadSchema().then((schema) => {
     var t = schema.getExpectedType(className, key);
-    var match = t ? t.match(/^relation<(.*)>$/) : false;
-    if (match) {
-      return match[1];
+    if (t.type == 'Relation') {
+      return t.targetClass;
```
Comment (drew-gross) [inline src/Controllers/DatabaseController.js]: nice

File: src/Schema.js [hunk 2]
```diff
@@ -363,6 +359,51 @@ class Schema {
...
+
+    for (let fieldName in defaultColumns[className]) {
+      fields[fieldName] = defaultColumns[className][fieldName];
+    }
+
+    let geoPoints = Object.keys(fields).filter(key => fields[key] && fields[key].type === 'GeoPoint');
+    if (geoPoints.length > 1) {
+      return {
+        code: Parse.Error.INCORRECT_TYPE,
+        error: 'currently, only one GeoPoint field may exist in an object. Adding ' + geoPoints[1] + ' when ' + geoPoints[0] + ' already exists.',
```
Comment (drew-gross) [inline src/Schema.js]: Since this is a mongo limitation and not a Parse Server limitation I think this logic should go into the mongo adapter

Comment (flovilmart) [inline src/Schema.js]: makes 200% sense

Comment (drew-gross) [inline src/Schema.js]: No, I think the adapter should return an error when you try to update the schema in a way that the adapter doesn't support. Then there is one less function that the adapter needs to implement.

Comment (flovilmart) [inline src/Schema.js]: Alrighty

**Comment by drew-gross** [comment]
This looks awesome! Feel free to merge right away if you want to do the geopoint thing in another PR.

**Comment by flovilmart** [comment]
I was definitely thinking of doing it separately as our update mechanic is not centralized in the MongoSchemaCollection.js

**Pull request merged**